clang/lib/Sema/SemaBounds.cpp

```diff
@@ -4017,11 +4023,55 @@ namespace {
         << ObservedBounds << ObservedBounds->getSourceRange();
     }
 
+    // BlameAssignmentWithinStmt prints a diagnostic message that highlights the
+    // assignment expression in St that causes V's observed bounds to be unknown
+    // or invalid.  If St is a DeclStmt, St itself and V are highlighted.
+    // BlameAssignmentWithinStmt returns the source location of the blamed
+    // assignment.
+    SourceLocation BlameAssignmentWithinStmt(Stmt *St, 
+                 const VarDecl *V, CheckingState State, unsigned DiagId) const {
+      SourceRange SrcRange = St->getSourceRange();
+      auto BDCType = Sema::BoundsDeclarationCheck::BDC_Statement;
+      
+      // For a declaration, the diagnostic message should start at the
+      // location of v rather than the beginning of St.  If the message
+      // starts at the beginning of a declaration T v = e, then extra
+      // diagnostics may be emitted for T.
+      SourceLocation Loc = St->getBeginLoc();
+      if (isa<DeclStmt>(St)) {
```

Comment:
Is is possible for a `DeclStmt` to have an entry in `State.BlameAssignments`? If not, you can early return here rather than searching for `V` in `BlameAssignments` below.

Reply:
`DeclStmt` does not have entries in `State.BlameAssignments`. I made the function return early inside the if block.